### PR TITLE
Remove globbing from all frontend code

### PR DESCRIPTION
**Pull request by philipp-spiess**
As @fkling noticed in #49684, these option is no longer supported:

- https://github.com/sourcegraph/sourcegraph/pull/27886
- https://github.com/sourcegraph/sourcegraph/pull/46045

Time to do some cleanup in the front end code!

Process:

- Grep for `globbing`
- Delete everything that is related to front end code

## Test plan

- CI

## App preview:

- [Web](https://sg-web-ps-rm-globbing-from-frontend.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| -0.01% (-0.32 kb) | -0.01% (-0.85 kb) | -0.00% (-0.53 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/ps/rm-globbing-from-frontend/compare-report.html;tab=live_object) for a full comparison between the commits 0f209d1 and b7d28f7 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Review by vovakulikov — APPROVED** [review]
Tested locally, code insights pickers and core search functionality works as expected, left small comments

File: client/shared/src/search/query/completion-utils.ts [hunk 1]
```diff
@@ -14,19 +14,17 @@ export const PREDICATE_REGEX = /^([.A-Za-z]+)\((.*?)\)?$/
  * regexInsertText escapes the provided value so that it can be used as value
  * for a filter which expects a regular expression.
  */
-export const regexInsertText = (value: string, options: { globbing: boolean }): string => {
-    const insertText = options.globbing ? value : `^${escapeRegExp(value)}$`
+export const regexInsertText = (value: string, _options: {}): string => {
```
[the remark this answers is not in the document]
Comment (philipp-spiess) [reply client/shared/src/search/query/completion-utils.ts:17]: yes

**Review by limitedmage — APPROVED** [review]
🚀

**Pull request merged**